### DISCUSSION TITLE
Add option to Archive sold cars

### DISCUSSION THREAD

**Discussion by kimma** [teslamate-org/teslamate#2736] [Ideas]
I've owned a few teslas (2) and they both still show up in the interface with status "Unavailable".
They also take up space on the various dashboards etc.

I suggest an option to "Archive" a sold vehicle so that it no longer tries fetching new data, and perhaps also hide the data in all dashboards unless specifically enabled ("Show archived vehicles" or something like that as a Grafana dropdown).

**Comment by g4ixt** [comment]
An associated option that would be nice is an archive function that moved some of the data for a current vehicle, based on dates, into a separate database that could be re-selected as an option.  My backup files are now >  500MiB which is not a problem but seems wasteful.

I have used SQL before but am not even remotely competent or confident with it, so even a list of the necessary steps to create and then use an archive database would be good.

**Reply by khorton** [reply]
I'd also love to have a simple way to archive data from a previous Tesla.  I could delete the data for that car from the database, but I haven't done that because very occasionally I want to check the date we drove to a specific location, etc.  I'd like an option to select which vehicles are displayed.  At the moment, I'm pondering spinning up a second TeslaMate instance from a backup, then deleting the new car data from that instance, then shutting it down.  I'd then delete the first Tesla data from the first TM instance.  If I ever need to see the old data, I'd start the TM instance for that car.

**Comment by totoantibes** [comment]
the benefit of archiving older data on smaller HW (like RPI) is that it would also run faster. I keep having timeouts as i have about 3 years of data on my teslamate.

**Comment by swiffer** [comment]
would like to implement such a feature (new column in cars table called hidden) - if set to true car is exluded in grafana dashboards.

@JakobLichterfeld - what do you think?

@totoantibes - what timeouts are you getting?

**Reply by JakobLichterfeld** [reply]
> @JakobLichterfeld - what do you think?

I do not have the issue on my side and I can imagine a improved UX if we add it. We do have a setting for disabling data collection (introduced by #3993), perhaps we can use this?

**Reply by swiffer** [reply]
i think it's separate concerns - maybe someone wants to stop collect new data but still show what has been collected so far.

**Reply by JakobLichterfeld** [reply]
Yeah, separate concerns, was just thinking about using the same switch, but you are right, separate setting would be better, but more settings means more potential points of failure of users as well.
At least the pr can be taken as a blueprint for setting option.

**Reply by swiffer** [reply]
While I'm at it - within `cars` table there currently is a column called `display_priority` which is not accessible via UI (db only). It has been introduced in 2021 for changing the order of cars in grafana.

https://github.com/teslamate-org/teslamate/pull/1904

- Should we make this one accessible via UI as well?
- Should the option to hide a car be in `cars` oder `car_settings` table? (prefer keeping them close together as both control dashboarding)?
- What should a new column be named? (display_in_dashboards (true|false))

Alternative approach to introducing a new column:

- Should we allow negative values in that column to exclude cars from dashboards? (e.g. `WHERE display_priority >= 1`)

**Reply by JakobLichterfeld** [reply]
> * Should we make this one accessible via UI as well?

So it is not used at all at the moment, innit?

> * Should the option to hide a car be in `cars` oder `car_settings` table? (prefer keeping them close together as both control dashboarding)?

My gutfeeling is both in settings

> * What should a new column be named? (display_in_dashboards (true|false))

Agree on display_in_dashboards

> * Should we allow negative values in that column to exclude cars from dashboards? (e.g. `WHERE display_priority >= 1`)

This is not as intuitive as a true or false imo.

**Reply by swiffer** [reply]
> > * Should we make this one accessible via UI as well?
> 
> So it is not used at all at the moment, innit?

I guess not many users are using it as it's not really visible. I'm making use of it since I know about it (from editing dashboards).
> 
> > * Should the option to hide a car be in `cars` oder `car_settings` table? (prefer keeping them close together as both control dashboarding)?
> 
> My gutfeeling is both in settings

Ok

> 
> > * What should a new column be named? (display_in_dashboards (true|false))
> 
> Agree on display_in_dashboards
> 

Ok

> > * Should we allow negative values in that column to exclude cars from dashboards? (e.g. `WHERE display_priority >= 1`)
> 
> This is not as intuitive as a true or false imo.

Ok

Will start implementation *soon 😄

**Reply by JakobLichterfeld** [reply]
> > > * Should the option to hide a car be in `cars` oder `car_settings` table? (prefer keeping them close together as both control dashboarding)?
> > 
> > 
> > My gutfeeling is both in settings
> 
> Ok

Wonder what you prefer? I do not have a clue why we have cars and car_settings

**Reply by swiffer** [reply]
Me neither... prefer keeping them together and therefore go with cars table  - or move both? car_settings have been introduced later...

**Reply by JakobLichterfeld** [reply]
Yeah, keeping them together. car_settings is for: if car needs to be locked to fall asleep , the deprecated sleep settings, streaming API. So having active and enabke_logging there seems not to be the wrong place.

**Comment by ultrixnoob** [comment]
I would really love to have the option to archive my old car ...
so what happened to this initiative?

**Reply by brianmay** [reply]
Somebody needs to do the work and create a PR...
